File: March_2022/30_03_2022/Find_fruits_left_over.py
from itertools import permutations
n = list(input())
x = ["a","m","o"]
while(True):
    flag = False
    for i in range(len(n)-2):
        if(sorted(n[i:i+3]) == x):
            del n[i:i+3]
            flag = True
            break
        
    if(flag == False):
        break

if(len(n) != 0):
    print("".join(n))
else:
    print("No fruit is left over")
    

File: March_2022/30_03_2022/Find_fruits_left_over.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int main(){

    string temp;
    cin>>temp;

    vector<char> n;
    for(int i=0;i<temp.length();i++){
        n.push_back(temp[i]);
    }

    vector<char> compare = {'a','m','o'};

    while(true){
        bool flag = false;
        for(int i=0;i<n.size()-2;i++){
            vector<char> sub_vector = {n.begin()+i,n.begin()+i+3};
            sort(sub_vector.begin(),sub_vector.end());
            if(sub_vector == compare){
                n.erase(n.begin()+i,n.begin()+i+3);
                flag = true;
                break;
            }
        }
        if(n.size()<3){
            break;
        }

        if(flag == false){
            break;
        }
    }

    if(n.size()>0){
        for(int i=0;i<n.size();i++){
            cout<<n[i];
        }
    }
    else{
        cout<<"No fruit is left over";
    }

    return 0;
}